File: src/main/java/org/eu/polarexpress/conductor/ConductorApplication.java
package org.eu.polarexpress.conductor;

import org.eu.polarexpress.conductor.config.DiscordConfig;
import org.eu.polarexpress.conductor.config.PixivConfig;
import org.eu.polarexpress.conductor.config.UtilConfig;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.properties.EnableConfigurationProperties;

@SpringBootApplication
@EnableConfigurationProperties({
        DiscordConfig.class,
        PixivConfig.class,
        UtilConfig.class
})
public class ConductorApplication {

    public static void main(String[] args) {
        SpringApplication.run(ConductorApplication.class, args);
    }

}

File: src/main/kotlin/org/eu/polarexpress/conductor/ConductorApplication.kt
package org.eu.polarexpress.conductor

import org.eu.polarexpress.conductor.config.DiscordConfig
import org.eu.polarexpress.conductor.discord.DiscordBot
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.boot.runApplication


@SpringBootApplication
@EnableConfigurationProperties(DiscordConfig::class)
class ConductorApplication

@Autowired
private val discordBot: DiscordBot? = null

fun main(args: Array<String>) {
	runApplication<ConductorApplication>(*args)
}
